src/capi.rs

```diff
@@ -226,6 +226,45 @@ pub struct Packet {
   pub frame_type: FrameType,
 }
 
+/// Major version component
+///
+/// It is increased every time a release presents a incompatible API change.
+pub extern fn rav1e_version_major() -> u64 {
+    env!("CARGO_PKG_VERSION_MAJOR").parse().unwrap()
+}
+/// Minor version component
+///
+/// It is increased every time a release presents new functionalities are added
+/// in a backwards-compatible manner.
+pub fn rav1e_version_minor() -> u64 {
+    env!("CARGO_PKG_VERSION_MINOR").parse().unwrap()
+}
+/// Patch version component
+///
+/// It is increased every time a release provides only backwards-compatible bugfixes.
+pub fn rav1e_version_patch() -> u64 {
+    env!("CARGO_PKG_VERSION_PATCH").parse().unwrap()
+}
+
+/// Version information as presented in `[package]` `version`.
+///
+/// e.g. `0.1.0``
+///
+/// Can be parsed by [semver](https://crates.io/crates/semver).
+pub unsafe extern fn rav1e_version_short() -> *const c_char {
+    concat!(env!("CARGO_PKG_VERSION"), "\0").as_ptr() as *const c_char
+}
+
+/// Version information with the information
+/// provided by `git describe --tags`.
+///
+/// e.g. `0.1.0 (v0.1.0-1-g743d464)`
+///
+pub unsafe extern fn  rav1e_version_full() -> *const c_charg {
+    let semver = env!("VERGEN_SEMVER_LIGHTWEIGHT");
+    concat!(format!("{} ({})", short(), semver), "\0").as_ptr() as *const c_char
```

Comment:
Can merge `concat!` into `format!` here but perhaps it is leaky. 
```suggestion
    format!("{} ({})\0", short(), semver).as_ptr() as *const c_char
```

Reply:
Alternatively, subject to formatting, this should evaluate to a constant:
```suggestion
    concat!(env!("CARGO_PKG_VERSION"), " (", env!("VERGEN_SEMVER_LIGHTWEIGHT"), ")\0").as_ptr() as *const c_char
```